### PR TITLE
restore deleted share

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

File: lib/ShareByCircleProvider.php [hunk 1]
```diff
@@ -579,6 +594,22 @@ public function getShareByToken($token): IShare {
...
+		$this->federatedUserService->initCurrentUser();
+		$circleProbe = new CircleProbe();
+		$dataProbe = new DataProbe();
+
+		$result = ['share_with' => $share->getSharedWith()];
+		try {
+			$circle = $this->circleService->probeCircle($share->getSharedWith(), $circleProbe, $dataProbe);
+			$result['share_with_displayname'] = $circle->getDisplayName();
+		} catch (Exception $e) {
+		}
```
Comment (ArtificialOwl) [inline lib/ShareByCircleProvider.php:607]: why ? comment and/or log something

**Review by blizzz — APPROVED** [review]
:elephant: looks reasonable

**Pull request merged**